### PR TITLE
feat(deployment): support private network

### PR DESCRIPTION
**Pull request by maikelordaz**
No changes in the contracts

Add configs for a private network deployed by Khalid and staging tests against that network, and minor refactor on deploy scripts

Tests made:
- check initialization variables
- create new term
- join term with yield generation
- start term
- pay contribution
- close funding period
- start new cycle
- withdraw fund
- withdraw collateral

Also added some mocks of the zaynfi contracts needed and their deploy scripts.

cc: @kkharji is tagged so he can check what he needs

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -7,6 +7,8 @@ DEPLOYER_ADDRESS=
 TESTNET_DEPLOYER_ADDRESS=
 # Mainnet RPC URL
 ARBITRUM_MAINNET_RPC_URL=
+# Private RPC URL
+TAKADAO_RPC_URL=
```
Comment (kkharji) [inline .env.example]: TAKADAO might be confusing a bit. How about `PRIVATENET_REMOTE_RPC_URL`, with ` takadao_private` renamed to `privatenet_remote`

Also if it's not too much work, `PRIVATENET_LOCAL_RPC_URL` to support local private network

**Review by MohammedTaka — COMMENTED** [review]
Looks like everything is completely mocked, good work. One question about multisig however. I don't think we need to do that on testnet, right?

File: .env.example (hunk 1 above)
Comment (maikelordaz) [reply .env.example]: Yes, no issue with that, changed it in commit 97a73d0

**Comment by maikelordaz** [comment]
> Looks like everything is completely mocked, good work. One question about multisig however. I don't think we need to do that on testnet, right?

@MohammedTaka yes, everything mocked so users can join with yield generation. and yes I agree the multisig is not needed on testnet

**Comment by elallamTaka** [comment]
@maikelordaz please resolve the conflicts, this needs to be merged

**Comment by maikelordaz** [comment]
@elallamTaka Conflicts solved

**Pull request merged**